stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -3201,33 +3197,23 @@ final internal class _Native${Self}StorageKeyNSEnumerator<
 }
 #endif
 
-/// This class is an artifact of the COW implementation.  This class only
-/// exists to keep separate retain counts separate for:
-/// - `${Self}` and `NS${Self}`,
-/// - `${Self}Index`.
-///
-/// This is important because the uniqueness check for COW only cares about
-/// retain counts of the first kind.
-///
-/// Specifically, `${Self}` points to instances of this class.  This class
-/// is also a proper `NS${Self}` subclass, which is returned to Objective-C
-/// during bridging.  `${Self}Index` points directly to
-/// `_Native${Self}Storage`.
-final internal class _Native${Self}StorageOwner<${TypeParametersDecl}>
+#if _runtime(_ObjC)
+/// This class exists for Objective-C bridging. It holds a reference to a
+/// `_Native${Self}Storage`, and can be upcast to `NS${Self}` when
+/// bridging is necessary.
+final internal class _Native${Self}BridgingStorage<${TypeParametersDecl}>
```

Comment:
@gribozavr, I see at least two possible complications:

* `_SwiftNativeNS${Self}` is a base class defined in the runtime. It won't be possible to have the storage impl inherit from both `ManagedBuffer` and that base class. (`_NS${Self}Core` is the shadow protocol that provides the functionality.)
* `_SwiftNativeNS${Self}` is only available when the Objective-C runtime is available.

Basically, the bridging storage needs to be cast-able to a `NSSet` or `NSDictionary`, and it needs space for `_heapBufferBridged_DoNotUse`, neither of which can be provided by a `ManagedBuffer` as far as I know.

The original double indirection implementation sidestepped the issue by having the storage owner (first indirection) serve as the bridged storage, and separating that from the managed buffer (second indirection).

I'll keep thinking about this. Worst comes to worst I can restore the double indirection; since the indices no longer refer to either object the data race bug that was mentioned in SE-0065 should at least not be an issue.